implementations/micrometer-registry-new-relic/src/main/java/io/micrometer/newrelic/NewRelicNamingConvention.java

```diff
@@ -33,16 +33,20 @@ public NewRelicNamingConvention(NamingConvention delegate) {
 
     @Override
     public String name(String name, Meter.Type type, @Nullable String baseUnit) {
-        return StringEscapeUtils.escapeJson(delegate.name(name, type, baseUnit));
+        return StringEscapeUtils.escapeJson(toValidNewRelicString(delegate.name(name, type, baseUnit)));
     }
 
     @Override
     public String tagKey(String key) {
-        return StringEscapeUtils.escapeJson(delegate.tagKey(key));
+        return StringEscapeUtils.escapeJson(toValidNewRelicString(delegate.tagKey(key)));
     }
 
     @Override
     public String tagValue(String value) {
-        return StringEscapeUtils.escapeJson(delegate.tagValue(value));
+        return StringEscapeUtils.escapeJson(toValidNewRelicString(delegate.tagValue(value)));
```

Comment:
Based on its doc, it seems the restriction is only for names, not values.

Reply:
Cool will fixup.